### PR TITLE
Improve Ginkgo device arch flags management.

### PR DESCRIPTION
**Pull request by tcojean**
Improve device arch flags and fix a few hwloc issues.

Arch flags related:
1. Use a newer version of CAS which makes Auto become All when no
   architecture was detected. This ensures that compiling on a node without
   a GPU will provide good performance (at the cost of binary size and
   compile time), instead of failing to provide optimized architecture
   specific kernels.
2. Use the `variable` version of CAS instead of `target` so that we do
   not call the CAS redundantly in several tests.
3. Fix HIP AMDGPU flags which were not properly passed:
   `target_compile_options` do not seem to work with `hip_add_library`, so
   the flags need to be passed to `hip_add_libary` (or executable)
   directly.
4. Compile only CUDA architecture sm_60 on Windows due to heap issues.

hwloc issues:
1. Fix inaccurate hwloc external version detection
2. Prevent rebuilding Ginkgo whenever calling CMake due to always copying
   the hwloc header and libs.

TODO:
+ [x] Update the CAS hash once it has been merged.

### CODE REVIEW

**Review by yhmtsai — APPROVED** [review]
LGTM. thanks for fixing it

File: hip/CMakeLists.txt [hunk 1]
```diff
@@ -240,10 +240,19 @@ if (GINKGO_HIP_PLATFORM MATCHES "nvcc")
         ARCHITECTURES ${GINKGO_CUDA_ARCHITECTURES}
         UNSUPPORTED "20" "21")
 endif()
+
+# `target_compile_options` do not work with hip_add_library
+# Thus, we need to pass the flags to `hip_add_library` itself
```
[the remark this answers is not in the document]
Comment (tcojean) [reply hip/CMakeLists.txt:245]: They already know about it:
https://github.com/ROCmSoftwarePlatform/rocALUTION/blob/develop/src/CMakeLists.txt#L119

**Review by upsj — APPROVED** [review]
LGTM!

**Comment by yhmtsai** [comment]
MSVC seems to work but blockjacobi runs out of heap

**Comment by tcojean** [comment]
I made MSVC compile only for CUDA architecture 60 to make sure it doesn't run out of heap, but we know from the previous windows pipeline that the new CAS otherwise works on Windows.

**Review by pratikvn — APPROVED** [review]
LGTM!

**Comment by tcojean** [comment]
Update: for convenience and to reduce the amount of PRs, this also now includes two fixes for HWLOC, one which comes from Tobias to prevent rebuilding Ginkgo whenever calling CMake due to always copying the hwloc header and libs.

**Pull request merged**